### PR TITLE
Match API reference to version of Kubernetes we're documenting

### PR DESCRIPTION
**Pull request by sftim**
Once backported appropriately, this could be a fix for issue #25548

### CODE REVIEW

**Comment by sftim** [comment]
@kbhawkey do you think this is an appropriate fix? Also, @zacharysarah I'd welcome your thoughts as you know the historical context here.

**Comment by zacharysarah** [comment]
@sftim The implementation looks solid in concept. I do notice that the "v" is already baked into the variable value.

https://5fd2ad7d6be32500089fbd14--kubernetes-io-master-staging.netlify.app/docs/reference/

Screen Shot 2020-12-10 at 3 58 57 PM

File: content/en/docs/reference/_index.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ This section of the Kubernetes documentation contains references.
 
 ## API Reference
 
-* [Kubernetes API Reference {{< latest-version >}}](/docs/reference/generated/kubernetes-api/{{< latest-version >}}/)
+* [API Reference for Kubernetes v{{< param "version" >}}](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/)
```
Comment (kbhawkey) [inline content/en/docs/reference/_index.md:21]: I think `version` and `latest` variables are equivalent?

Comment (tengqm) [reply content/en/docs/reference/_index.md:21]: IMO, `version` is the right param in this context.

Comment (kbhawkey) [reply content/en/docs/reference/_index.md:21]: Interesting. I looked at the previous configurations from 1.14 to 1.20.
The latest = version when the version is not deprecated?

Comment (kbhawkey) [reply content/en/docs/reference/_index.md:21]: Seems like v1.19, v1.18, v1.17 use the `latest` parameter to generated the API reference links.
I've opened PRs to modify the `latest` parameter to the deprecated version in these snapshots.
I agree that "version" makes more sense in this context.

Comment (sftim) [reply content/en/docs/reference/_index.md:21]: I'd expect that `latest` evaluated to the most recent released Kubernetes version, whatever documentation version you're looking at, whereas `version` evaluates to the version number for the documentation you are viewing.

Comment (sftim) [inline content/en/docs/reference/_index.md:21]:
```suggestion
* [API Reference for Kubernetes {{< param "version" >}}](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/)
```

Comment (kbhawkey) [reply content/en/docs/reference/_index.md]: Yes, though, how is this parameter used in the context of a deprecated version?

**Comment by kbhawkey** [comment]
@sftim . The change looks good:
https://deploy-preview-25552--kubernetes-io-master-staging.netlify.app/docs/reference/
Noted that the `latest-version` shortcode is used in other pages.

**Comment by kbhawkey** [comment]
Note: The `latest` parameter is used in these shortcodes:
latest-version
latest-semver
latest-release-notes
I can't recall when the `latest` parameter was introduced into the site configuration.

**Comment by sftim** [comment]
I'll PR a backport.

**Comment by kbhawkey** [comment]
I may have covered these changes in my recent PRs.

**Pull request merged**